jupytercad/notebook/objects/__init__.py

```diff
@@ -0,0 +1,106 @@
+from typing import Any, Dict, Optional, Union
+from pydantic import BaseModel, Extra
+from ..cad_document import CadDocument
+from ._schema.box import IBox
+from ._schema.cone import ICone
+from ._schema.cut import ICut
+from ._schema.cylinder import ICylinder
+from ._schema.extrusion import IExtrusion
+from ._schema.fuse import IFuse
+from ._schema.intersection import IIntersection
+from ._schema.jcad import Parts
+from ._schema.sphere import ISphere
+from ._schema.torus import ITorus
+
+
+class PythonJcadObject(BaseModel):
+    class Config:
+        arbitrary_types_allowed = True
+        underscore_attrs_are_private = True
+        extra = Extra.allow
+
+    name: str
+    shape: Parts
+    parameters: Union[
+        IBox,
+        ICone,
+        ICut,
+        ICylinder,
+        IExtrusion,
+        IIntersection,
+        IFuse,
+        ISphere,
+        ITorus,
+    ]
+
+    _caddoc = Optional[CadDocument]
+    _parent = Optional[CadDocument]
+
+    def __init__(__pydantic_self__, parent, **data: Any) -> None:   # noqa
+        super().__init__(**data)
+        __pydantic_self__._caddoc = CadDocument()
+        __pydantic_self__._caddoc.add_object(__pydantic_self__)
+        __pydantic_self__._parent = parent
+
+    def _repr_mimebundle_(self, **kwargs):
+        return self._caddoc.render()
+
+    def update_object(self):
```

Comment:
I'm wondering if this should be called automatically whenever the underlying parameters changed.

Instead of:
```python
box.parameters.Height = 1
box.update_object()
```
You would just do:
```python
box.parameters.Height = 1
```

This would be closer to the ipywidgets vision, if that is a goal we'd like to follow.

In ipywidgets, you can also hold changes so that we do not crowd the comm channel with too many messages:
```python
with widget.hold_sync():
    widget.value1 = 1
    widget.value2 = 2
    widget.value3 = 3
```

We could maybe think of having something equivalent here.

Reply:
I would love to do so, but `pydantic` does not have the field observation (I think @davidbrochart  is working on it). 
I use this approach in `JupyterCad` but we definitely should have something as you describe in the standalone version of `ypywidgets` frontend.

Reply:
See https://github.com/davidbrochart/ypywidgets/issues/6.